src/edit/handler/Edit.Poly.js

```diff
@@ -271,13 +281,13 @@ L.Edit.Poly = L.Handler.extend({
         }
 
         // create a ghost marker in place of the removed one
-        if (marker._prev && marker._next) {
+        if (marker._prev && marker._next && maxPoints > 0 && this._poly._latlngs.length < maxPoints) {
```

Comment:
`const lessThanMaxPoints = maxPoints > 0 && this._poly._latlngs.length < maxPoints`, or move into a method as luke suggested